src/190B20.c

```diff
@@ -1144,11 +1144,47 @@ INCLUDE_ASM(s32, "190B20", func_80263FE8);
 
 INCLUDE_ASM(s32, "190B20", func_80264084);
 
-INCLUDE_ASM(void, "190B20", add_xz_vec3f, Vec3f* vector, f32 speed, f32 angleDeg);
+void add_xz_vec3f(Vec3f* vector, f32 speed, f32 angleDeg) {
+    f32 newVectorZ;
+    f32 angleRad;
+    f32 sinAngleRad;
+
+    angleRad = angleDeg * 6.28318f;
+    angleRad /= 360.0f;
+    sinAngleRad = sin_rad(angleRad);
+    newVectorZ = vector->z + (-speed * cos_rad(angleRad));
+    angleRad = speed * sinAngleRad;
+    vector->x = vector->x + angleRad;
```

Comment:
any time you have x = x op y, you can write it as x op= y, so `vector->x += angleRad;`

That being said, I have a feeling there are temps for the sin_rad  and cos_rad and then the rest of the function is just 
```
vector->x += speed * sinAngleRad;
vector->z += speed * cosAngleRad;
```

could be wrong though, but it'd make this look a lot cleaner if it matches